se_apps/src/mainQt.cpp

```diff
@@ -226,43 +226,67 @@ int processAll(DepthReader *reader, bool processFrame, bool renderImages,
 	if (processFrame) {
 		Stats.start();
 	}
-  Eigen::Matrix4f pose;
+    Eigen::Matrix4f pose;
+	Eigen::Matrix4f gt_pose;
 	timings[0] = std::chrono::steady_clock::now();
-	if (processFrame && (reader->readNextDepthFrame(inputRGB, inputDepth))) {
-		frame = reader->getFrameNumber() - frameOffset;
-		if (powerMonitor != NULL && !firstFrame)
-			powerMonitor->start();
+	if (processFrame) {
 
-		timings[1] = std::chrono::steady_clock::now();
-		pipeline->preprocessing(inputDepth, 
-        Eigen::Vector2i(inputSize.x, inputSize.y), config->bilateralFilter);
+		// Read frames and ground truth data if set
+		bool read_ok;
+		if (config->groundtruth_file == "") {
+			read_ok = reader->readNextDepthFrame(inputRGB, inputDepth);
+		} else {
+			read_ok = reader->readNextData(inputRGB, inputDepth, gt_pose);
+		}
 
-		timings[2] = std::chrono::steady_clock::now();
+		// Process read frames 
+		if (read_ok) {
```

Comment:
Maybe nitpicking, wouldn't it be better here to either skip the iteration or terminate early instead of wrapping all the processing code in the if(read_ok) conditional?